src/joinboost/executor.py

```diff
@@ -404,6 +407,16 @@ def execute_spja_query(self,
         # group by and aggregate
         df = self.apply_group_by_and_agg(agg_conditions, df, group_by, window_by)
 
+        should_drop_columns = True
```

Comment:
maybe call it "projection push down"

Reply:
You can also apply it. In the worst case, it doesn't project out any attribute.
The check looks unnecessary.